### DISCUSSION TITLE
About pnpx deprecation

### DISCUSSION THREAD

**Discussion by ftzi** [pnpm/pnpm#5590] [General]
Having to type `pnpm dlx X` instead of just `pnpx X` really annoys me, as I like quicker stuff. It's obvious that 1~2s of extra typing isn't much but every single time I have to type it I feel it could just be shorter as efficiency/automation/simplicity matters a lot to me, to the point that it made me create this discussion here lol.

I imagine that you deprecated it to remove the exec/dlx ambiguity that it had, but honestly, we should have a short command to always execute `dlx`. `pnx`, `plx`, `pdlx`... honestly any short command would do it for me. I feel that the point of running remote packages is to fastly run stuff and `pnpm dlx X` goes contrary to this feeling. **Maybe we can just make `pnpx` to not run local packages in the next major.** `pnpm X` already executes a local package, and if the project has a script with the same package name (uncommon case), they can just use `pnpm exec` to be sure.

pnpm is already better than npm and yarn, I have no doubt. But this really small thing really annoys my daily usage of it.

**Comment by lvqq** [comment]
I have started a [project](https://github.com/lvqq/pnpm-alias) for pnpm alias rencently. If you are working with Windows Powershell/Command or zsh/bash, it may help.

**Reply by ftzi** [reply]
Cool but I insist that pnpx (or just pnx) should exist in pnpm.

**Reply by lvqq** [reply]
Yeah, i'm not sure if it will be builtin in the future😂

**Reply by zkochan** [reply]
pnpx is also not easy to type.

**Reply by ftzi** [reply]
I agree, but it's better than `pnpm dlx`. I guess `pnx` would do it greatly. `px` would also do it but it doesn't seem as clear as `pnx`. Also, notice that `pnx` is `npx` with p <-> n, so that's kinda cool!

**Reply by iamchathu** [reply]
Is there any update on this?

**Reply by ftzi** [reply]
I wish there was!

**Comment by angelhdzmultimedia** [comment]
You don't know how much this infuriates me. I opened a discussion and don't remember the outcome.

Every time Im installing some package and see the npx , muscular memory kicks in and I type pnpx, only to remember reading somewhere they made pnpx deprecated in favor of pnpm dlx. It's sooo silly, unnecessary, anti-DX.

**Reply by angelhdzmultimedia** [reply]
I will keep using pnpx. The moment it stops working as the alternative to npx, I'm gonna switch back to npm or to bun (bun still have issues with Nuxt).
 I started to promote pnpm and embraced it the first time I tried it, felt in love, made wallpapers, profile pictures, tutorials on YouTube.
 
 But I hate when people make decisions that are against DX,  just because...

Even Bun has bunx... 🙄

 I contributed with this section of adding an alias for pnpm in Windows Powershell... how ironic that there are still traces of **pnpx**... !!! 
![image](https://github.com/pnpm/pnpm/assets/10191085/e34d754b-125d-4574-89ee-a37811c294b3)

**Reply by ftzi** [reply]
I feel you, bro. I also loved pnpm and spread its message, and that also infuriates me. In my current project we use npm so I don't use pnpm for a long while already.

I intend to try bun when possible.

**Reply by angelhdzmultimedia** [reply]
Look what I found... 👀🤔🤯 https://github.com/antfu/ni
